app/models/host/managed.rb

```diff
@@ -848,6 +848,27 @@ def setup_clone
     @old = super { |clone| clone.interfaces = self.interfaces.map {|i| setup_object_clone(i) } }
   end
 
+  # rebuilds orchestration configuration for a host
+  # takes all the methods from Orchestration modules that are registered for configuration rebuild
+  # returns  : Hash with 'true' if rebuild was a success for a given key (Example: {"TFTP" => true, "DNS" => false})
+  def recreate_config
+    result = {}
+
+    Nic::Managed.rebuild_methods.map do |method, pretty_name|
+      interfaces.map do |interface|
+        value = interface.send method
+        result[pretty_name] = value if !result.has_key?(pretty_name) || (result[pretty_name] && !value)
+      end
+    end
+
+    self.class.rebuild_methods.map do |method, pretty_name|
+      raise ::Foreman::Exception.new(_("There are orchestration modules with methods for configuration rebuild
```

Comment:
Use N_() in combination with Foreman::Exception - we pass in the English string in order to generate a consistent error code, then translate it internally.  Also keep the string on one line as the i18n string extractor will think the \n is inside the string.

    ::Foreman::Exception.new(N_("There are.. %s"), pretty_name)

see http://projects.theforeman.org/projects/foreman/wiki/Translating#Exceptions